README.md
# libstress

This is an old project that should define a basic framework to do some stress tests. As of now there is only a little skeleton that may give you the basic idea but everything has to be written !

License is GPL v2 or higher.


NEWS
29.05.2009 
	Start of project libstress, a library for stress tests
	


autoclean.sh
#!/bin/bash
#
# Author : Olivier Delhomme
# Licence : GNU GPL v2
#
if [ -f Makefile ];
 then
    make distclean
fi;
rm -f compile
echo "rm -f compile"
rm -f aclocal.m4
echo "rm -f aclocal.m4"
rm -fr autom4te.cache
echo "rm -fr autom4te.cache"
rm -f config.guess
echo "rm -f config.guess"
rm -f config.sub
echo "rm -f config.sub"
rm -f depcomp
echo "rm -f depcomp"
rm -f INSTALL
echo "rm -f INSTALL"
rm -f install-sh
echo "rm -f install-sh"
rm -f ltmain.sh
echo "rm -f ltmain.sh"
rm -f missing
echo "rm -f missing"
rm -f src/*~ src/*#
echo "rm -f src/*~ src/*#"
rm -f *~ *#
echo "rm -f *~ *#"
rm -f Makefile.in src/Makefile.in
echo "rm -f Makefile.in src/Makefile.in"
rm -f configure
echo "rm -f configure"
rm -f config.status config.log
echo "rm -f config.status config.log"
rm -f libtool
echo "rm -f libtool"
find -name *~ -exec rm {} \;
echo "find -name *~ -exec rm {} \;"


src/libstress.c
/* -*- indent-tabs-mode: t; tab-width: 4; c-basic-offset: 4; -*- */
/*
   libstress.c
   libstress library

   (C) Copyright 2009 Olivier Delhomme
   e-mail : olivierdelhomme@gmail.com

   This program is free software; you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation; either version 2, or (at your option)
   any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License
   along with this program; if not, write to the Free Software Foundation,
   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
*/

#include "include/libstress.h"

/**
 *  Populates a new test_t structure with the right values
 *  @param groupname : a group_name where tests might be grouped together
 *  @param name : the test name
 *  @param dirname : the subdirectory name where the tests migth be conducted
 *  @param exec : the exec function for an atomic test (might be NULL but the test will not run)
 *  @param init : the init function for an atomic test (might be NULL)
 *  @param final: the final function for an atomic test (might be NULL)
 *  @param vary : the vary function used for internal datas (might be NULL)
 *  @param data : the internal datas (if any - might be NULL) for the test itself
 *  @param max_nb : maximum number of atomic test to be run
 *  @param max_thread : maximum threads to run at a time
 */
test_t *new_test_t_struct(gchar *groupname, gchar *name, gchar *dirname, gchar *exec_dirname, TestExecFunc exec, TestInitFunc init, TestFinalFunc final, TestVaryFunc vary, gpointer data, guint max_nb, guint max_threads)
{
	test_t *the_test = NULL;

	if (exec != NULL)
		{
			the_test = (test_t *) g_malloc0(sizeof(test_t));

			if (groupname == NULL)
				{
					the_test->groupname = g_strdup("Default Group");
				}
			else
				{
					the_test->groupname = g_strdup(groupname);
				}

			if (name == NULL)
				{
					the_test->name = g_strdup("Default Name");
				}
			else
				{
					the_test->name = g_strdup(name);
				}

			if (dirname == NULL)
				{
					the_test->dirname = g_strdup("dirname");
				}
			else
				{
					the_test->dirname = g_strdup(dirname);
				}

			if (exec_dirname == NULL)
				{
					the_test->exec_dirname = g_strdup("exec_dirname");
				}
			else
				{
					the_test->exec_dirname = g_strdup(exec_dirname);
				}

			the_test->exec = exec;
			the_test->init = init;
			the_test->final = final;
			the_test->vary = vary;

			the_test->data = data;

			the_test->max_nb = max_nb;
			the_test->max_threads = max_threads;

			the_test->percent = 0.0;  /**< at init time we do think that nothing happened within the test itself */

			return the_test;
		}
	else
		{
			return NULL;
		}
}


/**
 *  Adds a test to a test list
 *  @param test_list : a list of tests
 *  @param one_test : one test to add to the list
 */
GList *add_test_to_test_list(GList *test_list, test_t *one_test)
{
	if (one_test != NULL)
	{
		return g_list_append(test_list, (gpointer) one_test);
	}
	else
		return test_list;
}




po/update_languages.bash
#!/bin/bash
#    update_languages.bash
#    Updates all files for the translation system.
#
#    This file is part of "Sauvegarde" project.
#
#    (C) Copyright 2015 - 2016 Olivier Delhomme
#     e-mail : olivier.delhomme@free.fr
#
#    "Sauvegarde" is free software: you can redistribute it and/or modify
#    it under the terms of the GNU General Public License as published by
#    the Free Software Foundation, either version 3 of the License, or
#    (at your option) any later version.
#
#    "Sauvegarde" is distributed in the hope that it will be useful,
#    but WITHOUT ANY WARRANTY; without even the implied warranty of
#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#    GNU General Public License for more details.
#
#    You should have received a copy of the GNU General Public License
#    along with "Sauvegarde".  If not, see <http://www.gnu.org/licenses/>
#

rm -f libstress.po.old
mv libstress.po libstress.po.old

# Updating file list
./find_files.sh >./POTFILES.in

# Updating libstress.po
xgettext --language=C --default-domain=libstress --add-comments --from-code=UTF-8 --keyword=N_:1 --keyword=_:1 -F --files-from=./POTFILES.in

sed -i -e "s/CHARSET/us-ascii/" libstress.po

# updating translations
for l in fr; do
    msgmerge -U $l.po libstress.po;
done;

sed -i s,\.\.\/,, ./POTFILES.in


src/include/libstress.h
/* -*- indent-tabs-mode: t; tab-width: 4; c-basic-offset: 4; -*- */
/*
   libstress.h
   libstress library

   (C) Copyright 2009 Olivier Delhomme
   e-mail : olivierdelhomme@gmail.com

   This program is free software; you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation; either version 2, or (at your option)
   any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License
   along with this program; if not, write to the Free Software Foundation,
   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
*/
#ifndef _LIBSTRESS_H_
#define _LIBSTRESS_H_

#include <errno.h>
#include <fcntl.h>
#include <glib.h>
#include <stdio.h>
#include <stdlib.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <unistd.h>

#define LIBSTRESS_NAME "libstress"
#define LIBSTRESS_VERSION "0.0.1"
#define LIBSTRESS_SUMMARY "libstress library"
#define LIBSTRESS_DESCRIPTION "Provide some stress tests"
#define LIBSTRESS_AUTHOR "Olivier Delhomme <olivierdelhomme@gmail.com>"
#define LIBSTRESS_MAINTAINER "Olivier Delhomme <olivierdelhomme@gmail.com>"
#define LIBSTRESS_HOMEPAGE "none"
#define LIBSTRESS_DATE "29.05.2009"

#include "config.h"

/**
 *  Main structure
 */
typedef struct
{
	GList *test_list;   /**< lists of tests  */
} main_struct_t;


/**
 * Prototype definition for the tests functions
 * The vary function will be called until it returns FALSE
 * TRUE is passed when an initialisation is requiered (begin and end of the tests)
 */
typedef gboolean (* TestExecFunc) (gpointer, guint, guint); /**< Test execution function type          */
typedef void (* TestInitFunc) (gpointer);                   /**< Test initialisation function type     */
typedef void (* TestFinalFunc) (gpointer);                  /**< Test finalisation function type       */
typedef gboolean (* TestVaryFunc) (gpointer, gboolean);     /**< Test varying function definition type */


/**
 *  Test structure
 */
typedef struct
{
	gchar *groupname;       /**< test group name                                                     */
	gchar *name;            /**< test name                                                           */
	gchar *dirname;         /**< base directory name (should not change after init)                  */
	gchar *exec_dirname;    /**< directory name where the test should executes itself                */
	TestExecFunc exec;      /**< execution function for the specified test                           */
	TestInitFunc init;      /**< init function for the specified test                                */
	TestFinalFunc final;    /**< final function for the specified test                               */
	TestVaryFunc vary;      /**< vary funciton used to vary internal parameters                      */
	gpointer data;          /**< internal data of the test                                           */
	guint max_nb;           /**< max number of atomic test per threads                               */
	guint max_threads;      /**< max number of threads                                               */
	gfloat percent;         /**< values from 0.0 to 1.0 represents, in percent.                      */
} test_t;


/**
 *  Populates a new test_t structure with the right values
 *  @param groupname : a group_name where tests might be grouped together
 *  @param name : the test name
 *  @param dirname : the subdirectory name where the tests migth be conducted
 *  @param exec : the exec function for an atomic test (might be NULL but the test will not run)
 *  @param init : the init function for an atomic test (might be NULL)
 *  @param final: the final function for an atomic test (might be NULL)
 *  @param vary : the vary function used for internal datas (might be NULL)
 *  @param data : the internal datas (if any - might be NULL) for the test itself
 *  @param max_nb : maximum number of atomic test to be run
 *  @param max_thread : maximum threads to run at a time
 */
extern test_t *new_test_t_struct(gchar *groupname, gchar *name, gchar *dirname, gchar *exec_dirname, TestExecFunc exec, TestInitFunc init, TestFinalFunc final, TestVaryFunc vary, gpointer data, guint max_nb, guint max_threads);


/**
 *  Adds a test to a test list
 *  @param test_list : a list of tests
 *  @param one_test : one test to add to the list
 */
extern GList *add_test_to_test_list(GList *test_list, test_t *one_test);


#endif /* _LIBSTRESS_H_ */
